samples/panic_recover_wret.go

```diff
@@ -0,0 +1,33 @@
+// panic_recover_wret
```

Comment:
I have changed this sample to this: https://goplay.space/#RYWK7He_aTJ

This demonstrates the same thing but slightly differently:
1. `oddPanic` does what it says - it panics if the value is odd. Before it was capturing its own panic and just returning an error or not, so from the outside it was not panicking at all, even though it was named that it was. This way it is simpler.
2. The panic is recovered in a different function. This is more likely - I don't see much use case for recovering your own panic in a function - you may as well just return the error at the place panic is called.
3. The value passed to `panic` is an error - nicer than a string. I'd prefer not to encourage using strings for errors - maybe that's just me.